integration_tests/src/main/python/subquery_test.py

```diff
@@ -17,18 +17,108 @@
 from data_gen import *
 from marks import *
 
-gens = [('l', LongGen()), ('i', IntegerGen()), ('f', FloatGen()), (
-    's', StringGen()), ('d', decimal_gen_128bit)]
+@ignore_order(local=True)
+@pytest.mark.parametrize('data_gen', all_basic_gens, ids=idfn)
+def test_scalar_subquery_basics(data_gen):
+    assert_gpu_and_cpu_are_equal_sql(
+        lambda spark: gen_df(spark, [('a', data_gen)], num_slices=1),
```

Comment:
nit: Could you add a comment about why one slice is needed here? This applies to all of the tests that are being added.

Reply:
Done